lib/launcher.rb

```diff
@@ -127,9 +127,8 @@ def launch
 
       (t1 || t2).join
     rescue SignalException => e
-      # This is to prevent the exception handler below from catching SignalException exceptions.
       logger.info("Caught #{e}. Exiting")
-      raise
+      e.signo == 15 ? exit(143) : raise # trap SIGTERM, used by systemd to stop/restart the service
```

Comment:
> I wanted to retain the fact that it was terminated b/c of the signal. `143` is the exit code that shell will return for a process terminated by `SIGTERM` (that didn't handle the signal itself). 
> 
> Not sure how useful this behaviour is, I can return `0` instead.  

The usual practice is to return zero if that was a correct shutdown. But
143 seems misleading here (it's actually 128 + 15 which was number of
the signal in Bash). IMHO zero is more appropriate here.

-- 
Later,
 Lukas #lzap Zapletal